.github/workflows/_check-user.yml

```diff
@@ -0,0 +1,72 @@
+name: Check whether the workflow owner can use ARC runners
+
+on:
+  workflow_call:
+    inputs:
+      user_name:
+        required: true
+        type: string
+        description: The name of the workflow owner.
+      curr_branch:
+        required: true
+        type: string
+        description: Current branch.
+
+    outputs:
+      workflow-type:
+        description: Type of runners to use
+        value: ${{ jobs.check-user.outputs.workflow-type }}
+
+jobs:
+  check-user:
+    runs-on: linux.4xlarge
+    outputs:
+      workflow-type: ${{ steps.set-condition.outputs.workflow-type }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Check if user is in arc_users.yml
+        id: set-condition
+        run: |
+          curr_branch="${{ inputs.curr_branch }}"
+          echo "Current branch is '$curr_branch'"
+
+          if [[ "$curr_branch" == "nightly" || "$curr_branch" == "main" || "$curr_branch" == "release/"* || "$curr_branch" == "landchecks/"* ]]; then
+            echo "workflow-type=label" >> "$GITHUB_OUTPUT"
+          else
+            REPO_OWNER="pytorch"
+            REPO_NAME="test-infra"
+            ISSUE_NUMBER="5132"
+
+            COMMENTS=$(curl -H "Authorization: token $GITHUB_TOKEN" "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/issues/$ISSUE_NUMBER/comments")
+            USER_LIST=$(echo "$COMMENTS" | tr -d '\r\n' | jq '.[0].body' -r)
+            RUN_OPTION=$(echo "$COMMENTS" | tr -d '\r\n' | jq '.[1].body' --raw-output)
+
+            WORKFLOW_TYPE_RG="rg"
+            WORKFLOW_TYPE_LABEL="label"
+            WORKFLOW_TYPE_BOTH="both"
+
+            if [[ ${USER_LIST:0:1} == "!" ]]; then
+              # Disable ARC runners for everyone.
+              echo "workflow-type=$WORKFLOW_TYPE_LABEL" >> "$GITHUB_OUTPUT"
+            elif [[ ${USER_LIST:0:1} == "*" ]]; then
+              if [[ "$RUN_OPTION" == "$WORKFLOW_TYPE_BOTH" ]]; then
+                # Use ARC runners and old runners for everyone.
+                echo "workflow-type=$WORKFLOW_TYPE_BOTH" >> "$GITHUB_OUTPUT"
+              else
+                # Use only ARC runners for everyone.
+                echo "workflow-type=$WORKFLOW_TYPE_RG" >> "$GITHUB_OUTPUT"
+              fi
+            elif echo "$USER_LIST" | grep -q "${{ inputs.user_name }}"; then
```

Comment:
can this be made case insensitive? That's bitten us in the past occasionally

Reply:
But would case insensitivity be worse for usernames?

Reply:
Github usernames are not actually case sensitive:
https://stackoverflow.com/questions/58591331/how-to-change-the-case-of-my-githubs-username-without-changing-any-letter#:~:text=Usernames%20are%20not%20case%20sensitive,that%20Username%20is%20not%20available.